src/transformers/models/sam/modeling_sam.py

```diff
@@ -223,8 +223,11 @@ def _separate_heads(self, hidden_states: Tensor, num_attention_heads: int) -> Te
     def _recombine_heads(self, hidden_states: Tensor, point_batch_size: int) -> Tensor:
         batch, n_heads, n_tokens, c_per_head = hidden_states.shape
         hidden_states = hidden_states.transpose(1, 2)
+        # return hidden_states.reshape(
+        #     batch // max(1, point_batch_size), point_batch_size, n_tokens, n_heads * c_per_head
+        # )
         return hidden_states.reshape(
-            batch // max(1, point_batch_size), point_batch_size, n_tokens, n_heads * c_per_head
+            batch // point_batch_size, point_batch_size, n_tokens, n_heads * c_per_head
```

Comment:
@ArthurZucker @younesbelkada why would `point_batch_size` be < 1?

I am asking because this `max(1, point_batch_size)` is not traceable, so it ends up as a constant in the ONNX model which can lead to inference failure.

Reply:
This is a code legacy-related issue that needs to be addressed indeed. in the meta codebase they initialize the sparse embeddings with a point batch size of 0. We can just set it to 1 manually on the line 710 as there is a check later on to make sure these embeddings are not concatenated if they are empty (i.e. if their sum is different than 0)